package.json

```
@@ -1574,12 +1574,8 @@
                 ],
                 "configuration": "./languages/pip-requirements.json",
                 "filenamePatterns": [
-                    "*-constraints.txt",
-                    "*-requirements.in",
-                    "*-requirements.txt",
-                    "constraints-*.txt",
-                    "requirements-*.in",
-                    "requirements-*.txt"
+                    "**/*requirements{/**,*}.{txt,in}",
+                    "**/*constraints{/**,*}.txt"
```

We have to be careful to not over-fit on file patterns as we don't want to try and highlight files that are not requirements files.

```suggestion
                    "*-requirements.{txt, in}",
                    "*-contstraints.txt",
                    "requirements-*.{txt, in}",
                    "constraints-*.txt",
                    "requirements/*.{txt,in}",
                    "constraints/*.txt"
```

Let us know if this still highlights files as expected for you.

---

I made a counter suggestion.